Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2003-08-16
Initial Package Version: 0.2
Origin: Slackware Patch for Glibc
Description: Fixes Compile Errors from Glibc

diff -Naur ash-linux-0.2.orig/Makefile ash-linux-0.2/Makefile
--- ash-linux-0.2.orig/Makefile	1993-10-04 18:48:13.000000000 +0000
+++ ash-linux-0.2/Makefile	2003-08-16 20:50:59.000000000 +0000
@@ -14,8 +14,8 @@
 
 OBJS =	$(OBJ1) $(OBJ2)
 
-CFLAGS = -O2 -fomit-frame-pointer -m486 -DSHELL -I/usr/include/bsd -I.
-LDFLAGS = -s -lbsd
+CFLAGS = -O3 -pipe -fomit-frame-pointer -march=i686 -DSHELL -I. -D__BIT_TYPES_DEFINED__
+LDFLAGS = -s
 
 CLEANFILES =\
 	builtins.c builtins.h init.c mkinit mknodes mksyntax \
@@ -32,7 +32,7 @@
 token.def: mktokens
 	sh ./mktokens
 
-builtins.h builtins.c: mkbuiltins builtins
+builtins.h builtins.c: mkbuiltins
 	sh ./mkbuiltins
 
 init.c: mkinit $(SRCS)
diff -Naur ash-linux-0.2.orig/mknodes.c ash-linux-0.2/mknodes.c
--- ash-linux-0.2.orig/mknodes.c	1993-09-06 09:04:23.000000000 +0000
+++ ash-linux-0.2/mknodes.c	2003-08-16 20:50:59.000000000 +0000
@@ -89,7 +89,7 @@
 struct str *curstr;		/* current structure */
 
 
-FILE *infp = stdin;
+FILE *infp;
 char line[1024];
 int linno;
 char *linep;
@@ -102,6 +102,7 @@
 main(argc, argv)
 	char **argv;
 {
+	infp = stdin;
 	if (argc != 3) {
 		error("usage: mknodes file\n");
 		return(1);
diff -Naur ash-linux-0.2.orig/trap.c ash-linux-0.2/trap.c
--- ash-linux-0.2.orig/trap.c	1993-08-24 09:12:57.000000000 +0000
+++ ash-linux-0.2/trap.c	2003-08-16 20:50:59.000000000 +0000
@@ -53,6 +53,9 @@
 #include "mystring.h"
 #include <signal.h>
 
+#ifndef sig_t
+#define sig_t __sighandler_t
+#endif
 
 /*
  * Sigmode records the current value of the signal handlers for the various
